subsys/bluetooth/shell/bap_broadcast_assistant.c

```diff
@@ -373,32 +373,36 @@ static int cmd_bap_broadcast_assistant_add_src(const struct shell *sh,
 static int cmd_bap_broadcast_assistant_mod_src(const struct shell *sh,
 					       size_t argc, char **argv)
 {
-	int result;
 	struct bt_bap_broadcast_assistant_mod_src_param param = { 0 };
 	struct bt_bap_scan_delegator_subgroup subgroup = { 0 };
+	unsigned long src_id;
+	int err = 0;
+	int result;
 
-	param.src_id = strtol(argv[1], NULL, 0);
-	if (param.src_id < 0 || param.src_id > UINT8_MAX) {
+	src_id = shell_strtoul(argv[1], 0, &err);
+	if (err || src_id > UINT8_MAX) {
 		shell_error(sh, "adv_sid shall be 0x00-0xff");
 		return -ENOEXEC;
 	}
 
-	param.pa_sync = strtol(argv[2], NULL, 0);
-	if (param.pa_sync < 0 || param.pa_sync > 1) {
+	param.src_id = src_id;
+
+	param.pa_sync = shell_strtobool(argv[2], 0, &err);
+	if (err || param.pa_sync > 1) {
 		shell_error(sh, "pa_sync shall be boolean");
 		return -ENOEXEC;
 	}
 
 	if (argc > 3) {
-		param.pa_interval = strtol(argv[3], NULL, 0);
+		param.pa_interval = shell_strtoul(argv[3], 0, &err);
```

Comment:
Missing `err` check; you can either revert to using `strtol` or add the error check

Reply:
So, error check is needed only for `shell_sttoul()` ?

Reply:
Reverted